From 8b2d97eb56d6cffcc6ce235d6a6bf2d56055fe4b Mon Sep 17 00:00:00 2001
From: Michael Davis <mcarsondavis@gmail.com>
Date: Mon, 13 Nov 2023 09:24:24 -0500
Subject: [PATCH 044/188] Default 'roots' field of language config

Previously roots needed to be specified by every language and `[]` was
used as an explicit default. Root files don't make sense for every
language (for example TOML) so I think we should allow languages to
not explicitly mention the key and have the `[]` default automatically.
---
 helix-core/src/syntax.rs | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

--- a/helix-core/src/syntax.rs
+++ b/helix-core/src/syntax.rs
@@ -101,7 +101,8 @@ pub struct LanguageConfiguration {
     pub file_types: Vec<FileType>, // filename extension or ends_with? <Gemfile, rb, etc>
     #[serde(default)]
     pub shebangs: Vec<String>, // interpreter(s) associated with language
-    pub roots: Vec<String>,        // these indicate project roots <.git, Cargo.toml>
+    #[serde(default)]
+    pub roots: Vec<String>, // these indicate project roots <.git, Cargo.toml>
     pub comment_token: Option<String>,
     pub text_width: Option<usize>,
     pub soft_wrap: Option<SoftWrap>,
